great_expectations_cloud/agent/actions/compatibility.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+from typing import Protocol
+
+import great_expectations as gx
+from packaging.version import Version
+from packaging.version import parse as parse_version
+from typing_extensions import override
+
+
+class VersionRunner(Protocol):
+    def run_checkpoint(self) -> None:
+        ...
+
+    def run_check_datasource_config(self) -> None:
+        ...
+
+
+_VERSION_RUNNERS = {}
+
+
+def register_version_runner(version: str, runner: VersionRunner) -> None:
+    _VERSION_RUNNERS[version] = runner
```

Comment:
nit: I'd check version isn't already present and throw if it is so we don't accidentally overwrite.

Reply:
Good call, modified to:
```
def _register_version_runner(version: str, runner: _VersionRunner) -> None:
    if version in _VERSION_RUNNERS:
        raise ValueError(f"Version {version} already registered")
    _VERSION_RUNNERS[version] = runner
```
in https://github.dev/great-expectations/cloud/pull/148/files

I was also considering looking into registering when instantiating e.g. overriding `__new__`.